src/agent/coverage/src/block.rs

```diff
@@ -280,4 +343,61 @@ mod tests {
 
         assert_eq!(total, expected);
     }
+
+    #[test]
+    fn test_block_cov_serde() {
+        let block = BlockCov {
+            offset: 123,
+            count: 456,
+        };
+
+        let ser = serde_json::to_string(&block).unwrap();
+
+        let text = r#"{"offset":123,"count":456}"#;
+        assert_eq!(ser, text);
+
+        let de: BlockCov = serde_json::from_str(&ser).unwrap();
+        assert_eq!(de, block);
+    }
+
+    #[test]
+    fn test_cmd_cov_serde() {
+        #[cfg(target_os = "linux")]
+        const MAIN_EXE: &str = "/main.exe";
+
+        #[cfg(target_os = "linux")]
+        const SOME_DLL: &str = "/lib/some.dll";
+
+        #[cfg(target_os = "windows")]
+        const MAIN_EXE: &str = r"c:\main.exe";
+
+        #[cfg(target_os = "windows")]
+        const SOME_DLL: &str = r"c:\lib\some.dll";
+
+        let main_exe = module_path(MAIN_EXE);
+        let some_dll = module_path(SOME_DLL);
+
+        let cov = {
+            let mut cov = CommandBlockCov::default();
+            cov.insert(&main_exe, vec![1, 20, 300].into_iter());
+            cov.increment(&main_exe, 1);
+            cov.increment(&main_exe, 300);
+            cov.insert(&some_dll, vec![2, 30, 400].into_iter());
+            cov.increment(&some_dll, 30);
+            cov
+        };
+
+        let ser = serde_json::to_string(&cov).unwrap();
+
+        #[cfg(target_os = "linux")]
+        let text = r#"{"/lib/some.dll":[{"offset":2,"count":0},{"offset":30,"count":1},{"offset":400,"count":0}],"/main.exe":[{"offset":1,"count":1},{"offset":20,"count":0},{"offset":300,"count":1}]}"#;
```

Comment:
Serializing full property names feels unnecessarily verbose for a format that is essentially meant only for machines.

I think we'll see on the order of thousands of such instances, so the impact is neither trivial nor major, so I don't have a strong opinion on using something shorter.

Reply:
Yes, it could definitely be made even more terse. That said, there should only be one of these files per VM in the coverage task, and we will eventually support other output formats if needed. So I'll leave this for now.